app/user/tests/test_views.py

```diff
@@ -246,3 +247,104 @@ def test_invalid_user_id(self) -> None:
+
+
+class TestUserFollowingView(APITestCase):
+    def setUp(self) -> None:
+        self.password = fake.password()
+        self.user_one = User.objects.create_user(
+            username=fake.name(), email=fake.email(), password=self.password
+        )
+        self.user_two = User.objects.create_user(
+            username=fake.name(), email=fake.email(), password=self.password
+        )
+        self.client = APIClient()
+
+    @property
+    def bearer_token(self) -> Any:
+        login_url = reverse("login")
+        response = self.client.post(
+            login_url,
+            data={"email": self.user_two.email, "password": self.password},
+            format="json",
+        )
+        token = json.loads(response.content).get("access")
+        return {"HTTP_AUTHORIZATION": f"Bearer {token}"}
+
+    def test_authorized_get_followers(self) -> None:
+        url = reverse("following", kwargs={"id": self.user_one.id})
+        response = self.client.get(
+            url,
+            format="json",
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_unauthorized_get_followers(self) -> None:
+        url = reverse("following", kwargs={"id": self.user_one.id})
+        response = self.client.get(
+            url,
+            format="json",
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_authorized_user_follow(self) -> None:
+        url = reverse("follow")
+        response = self.client.post(
+            url,
+            data={"follow": self.user_two.id},
+            format="json",
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_authorized_user_unfollow(self) -> None:
+        url = reverse("follow")
+        response = self.client.post(
+            url,
+            data={"follow": self.user_two.id},
+            format="json",
+            **self.bearer_token,
+        )
+        url = reverse("unfollow", kwargs={"id": self.user_two.id})
+        response = self.client.delete(
+            url,
+            format="json",
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+    def test_unauthorized_user_unfollow(self) -> None:
+        url = reverse("unfollow", kwargs={"id": self.user_one.id})
+        response = self.client.delete(url, format="json")
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_unauthorized_user_follow(self) -> None:
+        url = reverse("follow")
+        response = self.client.post(
+            url,
+            data={"follow": self.user_one.id},
+            format="json",
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
```

Comment:
How is this one 201 and the user is unauthorized???? don't just write tests for the sake